### ISSUE TITLE
[Question] How to Bypass 64-bit Length Limit for Access Code in "Correlate Access Code - Tagged Stream - Fixed Length" Block?

### ISSUE THREAD

**Issue by Grifenfer** [paulgclark/gr-reveng#4]
### Description:
I'm using the "Correlate Access Code - Tagged Stream - Fixed Length" block in a GNU Radio flowgraph. I’ve encountered a limitation where the access code is restricted to a maximum length of 64 bits. My current project requires an access code longer than 64 bits.

### System Information:

    Operating System: Ubuntu 22.04.4 LTS
    GNU Radio Version: GNU Radio Companion 3.10.7.0

### Question:
Is there a way to extend or bypass the 64-bit length limit for the access code in this block? If not, are there any recommended alternatives or workarounds for handling longer access codes within GNU Radio?

### Request:
Any guidance, suggestions, or alternative approaches that can be used to handle access codes longer than 64 bits in this context would be greatly appreciated.

Thank you for your time and assistance.

**Comment by Grifenfer** [comment]
To bypass the 64-bit length limit for the access code in the "Correlate Access Code - Tagged Stream - Fixed Length" block, you can manually change the limit in the source code and then reinstall the module. Here’s how you can do it:

### Steps to Modify the Limit:

1. **Locate the Source File:**
   - Find the file `correlate_access_code_bb_ts_impl.cc` in the module's source code. This file is located in the `lib` directory of the module.

2. **Edit the Source Code:**
   - Open `correlate_access_code_bb_ts_impl.cc` in a text editor.
   - Go to line 88, where the 64-bit limit is defined.
   - Modify this limit to your desired number of bits. For example, if you want a 128-bit limit, you would change the relevant line of code accordingly.

   ```cpp
   // Original code (example, might vary slightly)
    d_len = access_code.length();	// # of bytes in string
      if(d_len > 64)
        return false;

   // Modified code for 128-bit limit
    d_len = access_code.length();	// # of bytes in string
      if(d_len > 128)
        return false;
   ```

The comment says that it is the number of bytes but it the number of bits.

3. **Reinstall the Module:**
   - After making the change, you need to rebuild and reinstall the module. 

4. **Verify the Changes:**
   - Once reinstalled, test the block in your GNU Radio flowgraph to ensure that it now accepts access codes up to the new limit you've set.

---

By following these steps, you should be able to bypass the default 64-bit limit for the access code in the "Correlate Access Code - Tagged Stream - Fixed Length" block.

**Comment by paulgclark** [comment]
Hello and thank you for the feedback.

Full disclosure - I've moved away from using this block in recent years in favor of these two gr-satellites blocks:
Sync and Create PDU
Sync and Create Packed PDU

I'll update the source here with your fix, however, for those who still like to use this block.

**Comment by Grifenfer** [comment]
Hello and thank you for your answer. 

Where can I find the gr-satellites module ?

**Comment by paulgclark** [comment]
https://github.com/daniestevez/gr-satellites

**Comment by paulgclark** [comment]
Hi - I took a few minutes to implement a fix for this and saw where the 64 is coming from - it's a the size of:
`unsigned long long d_access code` (defined in `correlate_access_code_bb_ts_impl.h`) 

I'm not sure of a clean way to increase the size of this variable past 64 bits. This code was forked from the Correlate Access Code - Tag Stream in the main tree and I don't want to do a major rewrite when the satellites implementation is out there.

Closing as a no-fix, the 64-bit limitation will not affect most users, I think. 

Thanks again for bringing this to my attention, though.
